contracts/DCNTCrescendo.sol

```diff
@@ -129,8 +133,17 @@ contract DCNTCrescendo is
     return ((bps - takeRateBPS) * _currentPrice[id]) / bps;
   }
 
-  function buy(uint256 id) external payable {
-    require(saleIsActive, "Sale must be active to buy");
+  modifier salesAreActive {
+    require(block.timestamp >= saleStart, "Sales are not active yet.");
+    require(! saleIsPaused, "Sale must be active to buy or sell");
+    _;
+  }
```

Comment:
I feel like we are duplicating code between this modifier and `saleIsActive` can we dry this up and have this modifier call `saleIsActive`?